src/libraries/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/CompilerServices/NewLateBinding.vb

```diff
@@ -927,6 +942,18 @@ Namespace Microsoft.VisualBasic.CompilerServices
                 baseReference = New Container(Instance)
             End If
 
+            If baseReference.IsCOMObject And Not baseReference.IsWindowsRuntimeObject Then
+                Try
+                    LateBinding.InternalLateSet(Instance, Type, MemberName, Arguments, ArgumentNames, OptimisticSet, CallType)
+                    If (RValueBase And Type.IsValueType) Then
```

Comment:
>If (RValueBase And Type.IsValueType) Then [](http://example.com/codeflow?start=20&length=41)

Should the `If` block be before `LateBinding.InternalLateSet()`? (Is there a reason to call the `InternalLateSet()` if the we'll throw an exception afterwards?) And consider moving the `If` block outside of the `Try / Catch`.

Reply:
This is needed because `Type` is byref and can be updated.